contracts/deployment_scripts/messenger/layer2/001_deploy_cross_chain_messenger.ts

```diff
@@ -0,0 +1,34 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
+
+/* 
+    This script deploys the L2 side of the cross chain messenger.
+    It requires the address of the message bus on the L2 side, which is generated by the enclave
+    on genesis. 
+*/
+
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+    const { 
+        deployments, 
+        getNamedAccounts
+    } = hre;
+
+    // Get the prefunded L2 deployer account to use for deploying.
+    const {deployer} = await getNamedAccounts();
+
+    // TODO: Remove hardcoded L2 message bus address when properly exposed.
+    const busAddress = hre.ethers.utils.getAddress("0x526c84529b2b8c11f57d93d3f5537aca3aecef9b")
```

Comment:
I think this is now possible to be hooked up right ?

Reply:
thats the l2 message bus that the enclave deploys on genesis. It's not possible yet to hook it up, there is a todo on it to finish up as its currently generated with a pre hardcoded key.

Reply:
(i need to hook it up to whatever the process for encrypting rollups and using the secret ends up being)